### PR TITLE
Adding ability to change rendering on the fly

### PR DESCRIPTION
**Pull request by rutaihwa**
Enable user to add custom renderings

### CODE REVIEW

File: src/main/scala/net/kaliber/pdf/PdfRenderer.scala [hunk 1]
```diff
@@ -1,23 +1,23 @@
...
 import org.xhtmlrenderer.pdf.ITextRenderer
 import org.xhtmlrenderer.resource.XMLResource
 import org.xhtmlrenderer.context.StyleReference
+import org.xhtmlrenderer.extend.FontResolver
 
 /**
  * Simple wrapper.
  *
- * @classLoader The class loader used to resolve assets
+ * `PdfRenderer` wraps the the rendering with iText
```
Comment (EECOLOR) [inline src/main/scala/net/kaliber/pdf/PdfRenderer.scala]: What does this mean? I do not think this helps endusers of the library.

File: src/test/scala/Test.scala [hunk 2]
```diff
@@ -1,15 +1,22 @@
...
+import org.xhtmlrenderer.pdf.ITextRenderer
+
 import scala.io.Source
 
 object Test extends Specification {
 
-  val examples = Paths.get("./example")
-  val `test.pdf` = examples resolve "test.pdf"
+  val examples: Path = Paths.get("./example")
+  val `test.pdf`: Path = examples resolve "test.pdf"
```
Comment (EECOLOR) [inline src/test/scala/Test.scala:13]: Just curious, why did you add the type declarations?

File: src/test/scala/Test.scala [hunk 3]
```diff
@@ -1,15 +1,22 @@
...
+
+  // Font should be rendered from anywhere
+  val rendererWithFonts:ITextRenderer = new ITextRenderer
+  rendererWithFonts.getFontResolver.addFontDirectory(examples.toAbsolutePath.toString + "/fonts", true)
 
-s"""|============================
+  s"""|============================
```
Comment (EECOLOR) [inline src/test/scala/Test.scala:19]: If you indent this line, you should also indent the ones below. Make sure the `|` chars line out. I do not think the indentation is an improvement, but that's taste. So do with it what you want 😄

File: src/test/scala/Test.scala [hunk 4]
```diff
@@ -24,12 +31,15 @@ s"""|============================
...
-       val renderer = new PdfRenderer(classLoader)
+       val renderer = new PdfRenderer(classLoader, rendererWithFonts)
        renderer.toBytes(body)
      }
 
      Files write (`test.pdf`, pdf)
 
      todo
+
+    // Todo Write proper tests please
```
Comment (EECOLOR) [inline src/test/scala/Test.scala:42]: Although I admire your intent. Writing tests for something that wraps another library which results in a side effect (pdf) is probably more trouble than it's worth.

No matter what happens, a human needs to check the resulting PDF on every code change or dependency update.

I am however interested in your thoughts on 'proper tests' here.

**Review by EECOLOR — APPROVED** [review]
Great work! Thank you for creating a pull request!

I have some comments, but none that are blocking.

File: src/test/scala/Test.scala (hunk 4 above)
Comment (rutaihwa) [reply src/test/scala/Test.scala:42]: You are right, its so much work for what's worth. But I was thinking more to test more the elements that we expect to be there on a document.

File: src/test/scala/Test.scala (hunk 3 above)
Comment (rutaihwa) [reply src/test/scala/Test.scala:19]: Just taste.

File: src/main/scala/net/kaliber/pdf/PdfRenderer.scala (hunk 1 above)
Comment (rutaihwa) [reply src/main/scala/net/kaliber/pdf/PdfRenderer.scala]: When I started to read the code I had put some comments for me to understand, I must've stayed on this mindset when I left the comment. :frowning:

File: src/test/scala/Test.scala (hunk 2 above)
Comment (rutaihwa) [reply src/test/scala/Test.scala:13]: It was just for readability, nothing more.

**Comment by EECOLOR** [comment]
:+1:  As soon as I have time I will merge and publish it, I hope I can find a moment tomorrow.

**Comment by rutaihwa** [comment]
Great! :+1 This is a nice library.

**Comment by EECOLOR** [comment]
@rutaihwa Thank you for your contribution!
